ipaserver/rpcserver.py

```diff
@@ -72,6 +73,10 @@
 if six.PY3:
     unicode = str
 
+# time.perf_counter_ns appeared in Python 3.7.
+if version_info < (3, 7):
+    time.perf_counter_ns = lambda: int(time.perf_counter() * 10**9)
```

Comment:
Just a note: the version check is not necessary here, the `feature` check is preferred.